### PR TITLE
feat(calling): request audio and BT permission on start, join or accept call (WPB-1809) (WPB-3961)

### PR DESCRIPTION
**Pull request by ohassine**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

This PR aims to request audio/BT permission on every action related to calling(join, answer or start)

If the user decline the permission twice, it will be considered as permanent deny and we will display a dialog containing a button to redirect to app info in settings

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

Try to start a call and reject permissions every time it's requested

### Attachments (Optional)

Wire 2023-08-31 at 2_50 PM

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [443](https://10.10.124.134/job/AR-build-pipeline/443/) **failed**.

**Comment by AndroidBob** [comment]
Build [445](https://10.10.124.134/job/AR-build-pipeline/445/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7452606)(7452606)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7452606)(7452606)-staging-compat.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/recordaudio/RecordAudioComponent.kt [hunk 1]
```diff
@@ -145,7 +146,10 @@ fun RecordAudioComponent(
         dialogState = viewModel.getPermissionsDeniedDialogState(),
         onDismiss = viewModel::onDismissPermissionsDeniedDialog,
         onOpenSettings = {
-            context.startActivity(Intent(Settings.ACTION_SETTINGS))
+            val intent = Intent(Settings.ACTION_APPLICATION_DETAILS_SETTINGS)
+            val uri = Uri.fromParts("package", context.packageName, null)
+            intent.setData(uri)
+            context.startActivity(intent)
```
Comment (alexandreferris) [inline app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/recordaudio/RecordAudioComponent.kt]: 🙌🏻

**Review by alexandreferris — APPROVED** [review]
amazing work!

File: app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/recordaudio/RecordAudioComponent.kt (hunk 1 above)
Comment (ohassine) [reply app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/recordaudio/RecordAudioComponent.kt]: Good catch, missed that

**Comment by AndroidBob** [comment]
Build [460](https://10.10.124.134/job/AR-build-pipeline/460/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7459435)(7459435)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7459435)(7459435)-staging-compat.apk)

**Review by Garzas — APPROVED** [review]

**Review by yamilmedina — APPROVED** [review]

**Pull request merged**